extensions/notebook/src/book/bookModel.ts

```diff
@@ -0,0 +1,231 @@
+						title: sections[i].title,
+						root: root,
+						tableOfContents: tableOfContents,
+						page: sections[i],
+						type: BookTreeItemType.ExternalLink,
+						treeItemCollapsibleState: vscode.TreeItemCollapsibleState.Collapsed,
+						isUntitled: this.openAsUntitled
+					},
+						{
+							light: this._extensionContext.asAbsolutePath('resources/light/link.svg'),
+							dark: this._extensionContext.asAbsolutePath('resources/dark/link_inverse.svg')
+						}
+					);
+
+					notebooks.push(externalLink);
+				} else {
+					let pathToNotebook = path.join(root, 'content', sections[i].url.concat('.ipynb'));
+					let pathToMarkdown = path.join(root, 'content', sections[i].url.concat('.md'));
+					// Note: Currently, if there is an ipynb and a md file with the same name, Jupyter Books only shows the notebook.
+					// Following Jupyter Books behavior for now
+					if (await fs.pathExists(pathToNotebook)) {
+						let notebook = new BookTreeItem({
+							title: sections[i].title,
+							root: root,
+							tableOfContents: tableOfContents,
+							page: sections[i],
+							type: BookTreeItemType.Notebook,
+							treeItemCollapsibleState: vscode.TreeItemCollapsibleState.Collapsed,
+							isUntitled: this.openAsUntitled
+						},
+							{
+								light: this._extensionContext.asAbsolutePath('resources/light/notebook.svg'),
+								dark: this._extensionContext.asAbsolutePath('resources/dark/notebook_inverse.svg')
+							}
+						);
+
+						if (this.openAsUntitled) {
+							if (!this._allNotebooks.get(path.basename(pathToNotebook))) {
+								this._allNotebooks.set(path.basename(pathToNotebook), notebook);
+								notebooks.push(notebook);
+							}
+						}
+						else {
+							if (!this._allNotebooks.get(pathToNotebook)) {
+								this._allNotebooks.set(pathToNotebook, notebook);
+								notebooks.push(notebook);
+							}
+						}
+					} else if (await fs.pathExists(pathToMarkdown)) {
+						let markdown = new BookTreeItem({
+							title: sections[i].title,
+							root: root,
+							tableOfContents: tableOfContents,
+							page: sections[i],
+							type: BookTreeItemType.Markdown,
+							treeItemCollapsibleState: vscode.TreeItemCollapsibleState.Collapsed,
+							isUntitled: this.openAsUntitled
+						},
+							{
+								light: this._extensionContext.asAbsolutePath('resources/light/markdown.svg'),
+								dark: this._extensionContext.asAbsolutePath('resources/dark/markdown_inverse.svg')
+							}
+						);
+						notebooks.push(markdown);
+					} else {
+						let error = localize('missingFileError', "Missing file : {0}", sections[i].title);
+						vscode.window.showErrorMessage(error);
+					}
+				}
+			} else {
+				// TODO: search functionality (#6160)
+			}
+		}
+		return notebooks;
+	}
+
+	/**
+	 * Recursively parses out a section of a Jupyter Book.
+	 * @param section The input data to parse
+	 */
+	private parseJupyterSection(section: any[]): IJupyterBookSection[] {
```

Comment:
>parseJupyterSection [](start = 9, length = 19)

Shouldn't this be parseJupyterSections? (plural) It's parsing out possibly multiple sections so that would make more sense

Reply:
well I merged this over from your PR for open specific book. I haven't actually taken given a thought. I can change the name